### PR TITLE
Fix swap max price calculation

### PR DESCRIPTION
**Pull request by bvbfan**
Fixes https://github.com/cakedefi/defichain-private/issues/404

### CODE REVIEW

**Review by monstrobishi — APPROVED** [review]

File: src/masternodes/mn_rpc.cpp [hunk 1]
```diff
@@ -2877,9 +2877,18 @@ void CheckAndFillPoolSwapMessage(const JSONRPCRequest& request, CPoolSwapMessage
...
             auto resA = pool.reserveA;
             auto resB = pool.reserveB;
-            if (poolSwapMsg.idTokenFrom != pool.idTokenA) {
-                std::swap(resA, resB);
+
+            // NOTE it has a bug prior Dakota hardfork
+            if (height < Params().GetConsensus().DakotaHeight) {
+                if (poolSwapMsg.idTokenFrom != pool.idTokenA) {
+                    std::swap(resA, resB);
+                }
```
Comment (monstrobishi) [inline src/masternodes/mn_rpc.cpp:2885]: We don't need hardfork check on RPC side

Comment (monstrobishi) [reply src/masternodes/mn_rpc.cpp:2885]: If maxPrice is null just set both to int MAX, then it has no protection @uzyn asked, then the app can pass in max price

Comment (bvbfan) [reply src/masternodes/mn_rpc.cpp]: Done.

**Review by ShengguangXiao — APPROVED** [review]

**Comment by ShengguangXiao** [comment]
`feature_poolswap.py` failed on travis.

**Pull request merged**